src/ui/public/markdown/markdown.js

```diff
@@ -56,70 +60,46 @@ export function markdownFactory(whiteListedRules, openLinksInNewTab = false) {
       return originalLinkRender(tokens, idx, options, env, self);
     };
   }
-
-  return markdownIt;
-}
-
-export class Markdown extends Component {
-  constructor(props) {
-    super(props);
-
-    this.markdownIt = markdownFactory(this.props.whiteListedRules, this.props.openLinksInNewTab);
-
-    this.state = {
-      renderedMarkdown: this.transformMarkdown(this.props),
-    };
-  }
-
   /**
-   * This method is used to actually render markdown from the passed parameter
+   * This method is used to render markdown from the passed parameter
    * into HTML. It will just return an empty string when the markdown is empty.
-   * Since we want to use this with dangerouslySetInnerHTML, this method returns
-   * the required object format with an __html key in it.
+   * @param {String} markdown - Rhe markdown String
+   * @return {Object} - Returns the dangerouslySetInnerHTML required object
+   * with the rendered HTML stored in the __html key.
    */
-  transformMarkdown(params) {
-    if (!params.markdown) {
+  return (markdown) => {
+    if (!markdown) {
       return { __html: '' };
     }
-    return { __html: this.markdownIt.render(params.markdown) };
-  }
-
-  componentWillReceiveProps(props) {
-    const hasOpenLinksInNewTabChanged = props.openLinksInNewTab !== this.props.openLinksInNewTab;
-    const hasMarkdownChanged = props.markdown !== this.props.markdown;
-    const hasWhiteListerRulesChanged = props.whiteListedRules !== this.props.whiteListedRules;
-
-    if (hasOpenLinksInNewTabChanged || hasWhiteListerRulesChanged) {
-      this.markdownIt = markdownFactory(props.whiteListedRules, props.openLinksInNewTab);
-    }
-    if (hasMarkdownChanged || hasOpenLinksInNewTabChanged || hasWhiteListerRulesChanged) {
-      this.setState({
-        renderedMarkdown: this.transformMarkdown(props),
-      });
-    }
-  }
+    return { __html: markdownIt.render(markdown) };
```

Comment:
I think we should rather return only the rendered markdown from this function, and wrap it in `{ __html }` in the component below. That way the vanilla JS function stays less related to React.

Reply:
Ok will do ;)